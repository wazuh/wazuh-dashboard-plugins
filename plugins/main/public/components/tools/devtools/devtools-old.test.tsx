import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
// Import of module under test will be done after mocks

// Mocks
jest.mock('../../common/hocs', () => ({
  // Identity HOC so we render the wrapped component without referencing React
  withGlobalBreadcrumb:
    () =>
    (Component: any): React.FC<any> =>
    props =>
      Component(props),
  __esModule: true,
}));

jest.mock('../../../kibana-services', () => ({
  getUiSettings: jest.fn(() => ({
    get: jest.fn().mockImplementation((key: string) => {
      if (key === 'home:useNewHomePage') return false;
      return false;
    }),
  })),
  __esModule: true,
}));

jest.mock('../../../utils/applications', () => ({
  devTools: {
    id: 'dev-tools',
    breadcrumbLabel: 'Dev Tools',
  },
  __esModule: true,
}));

jest.mock('./application/hooks/use-setup', () => ({
  __esModule: true,
  default: () => ({
    editorInputRef: {
      current: {
        /* minimal stub */
      },
    },
    editorOutputRef: { current: { getValue: jest.fn(() => '{}') } },
  }),
}));

jest.mock('./lib', () => ({
  __esModule: true,
  send: jest.fn(),
  saveEditorContentAsJson: jest.fn(),
}));

// jQuery ($) stub for the hotkey hook used by the component
const handlers: Record<string, any> = {};
const on = jest.fn((evt: string, cb: Function) => {
  handlers[evt] = cb;
});
const off = jest.fn((evt: string) => {
  delete handlers[evt];
});
(global as any).$ = () => ({ on, off });

// Require after mocks and globals are in place
const lib = require('./lib');
const { ToolDevTools } = require('./devtools-old');
const { DEV_TOOLS_BUTTONS } = require('./constants');

// Integration tests for the DevTools console container
describe('ToolDevTools (integration)', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('clicking Play sends the request and updates status meta (happy path)', async () => {
    // Arrange mocked send to invoke hooks
    lib.send.mockImplementation(
      (
        _in: any,
        _out: any,
        _first: boolean,
        hooks?: { onStart?: Function; onEnd?: Function },
      ) => {
        hooks?.onStart?.();
        hooks?.onEnd?.({
          ok: true,
          status: 200,
          statusText: 'OK',
          durationMs: 123.4,
        });
      },
    );

    const { container } = render(<ToolDevTools />);

    // Act: click Play
    const play = container.querySelector(
      `#${DEV_TOOLS_BUTTONS.PLAY_BUTTON_ID}`,
    );
    expect(play).toBeTruthy();
    await act(async () => {
      fireEvent.click(play!);
    });

    // Assert send called and UI reflects status
    expect(lib.send).toHaveBeenCalledTimes(1);
    expect(screen.getByText(/200\s-\sOK/i)).toBeInTheDocument();
    expect(screen.getByText(/123\s?ms/i)).toBeInTheDocument();
  });

  it('clicking Export triggers saveEditorContentAsJson (edge case: no meta yet)', () => {
    const { container } = render(<ToolDevTools />);

    // Export button provided by top nav config
    const exportBtn = container.querySelector(
      '[data-test-subj="consoleExportButton"]',
    );
    expect(exportBtn).toBeTruthy();
    fireEvent.click(exportBtn!);

    expect(lib.saveEditorContentAsJson).toHaveBeenCalledTimes(1);
  });

  it('handles error meta from send (status not ok)', async () => {
    lib.send.mockImplementation(
      (
        _in: any,
        _out: any,
        _first: boolean,
        hooks?: { onStart?: Function; onEnd?: Function },
      ) => {
        hooks?.onStart?.();
        hooks?.onEnd?.({
          ok: false,
          status: 500,
          statusText: 'Internal Error',
          durationMs: 45.2,
        });
      },
    );

    render(<ToolDevTools />);

    // Click Play
    const play = document.querySelector(`#${DEV_TOOLS_BUTTONS.PLAY_BUTTON_ID}`);
    await act(async () => {
      fireEvent.click(play!);
    });

    // Error badge shown
    expect(screen.getByText(/500\s-\sInternal Error/i)).toBeInTheDocument();
  });
});
