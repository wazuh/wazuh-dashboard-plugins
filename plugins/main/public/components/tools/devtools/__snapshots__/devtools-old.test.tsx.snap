// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolDevTools (integration) renders and matches snapshot (initial state) 1`] = `
<div>
  <div
    class="euiTabs euiTabs--small"
    role="tablist"
  >
    <button
      aria-selected="true"
      class="euiTab euiTab-isSelected"
      role="tab"
      type="button"
    >
      <span
        class="euiTab__content"
      >
        Console
      </span>
    </button>
  </div>
  <div
    class="euiFlexGroup euiFlexGroup--directionColumn euiFlexGroup--responsive wz-dev-tools"
    style="padding: 8px; margin: 0px; --col-separator-width: 14px; --col-left-width: 46.5%; --col-right-width: calc(100% - var(--col-left-width));"
  >
    <div
      class="euiFlexItem"
    >
      <div
        class="euiFlexGroup euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiTabs euiTabs--small"
            role="tablist"
          >
            <button
              aria-selected="false"
              class="euiTab"
              data-test-subj="consoleExportButton"
              role="tab"
              title="Export"
              type="button"
            >
              <span
                class="euiTab__content"
              >
                Export
              </span>
            </button>
            <button
              aria-selected="false"
              class="euiTab"
              data-test-subj="consoleApiReferenceButton"
              role="tab"
              title="API Reference"
              type="button"
            >
              <span
                class="euiTab__content"
              >
                API Reference
              </span>
            </button>
          </div>
        </div>
        <div
          class="euiFlexItem"
        >
          <div
            class="euiFlexGroup euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
            style="gap: 8px;"
          />
        </div>
      </div>
      <div
        class="euiFlexGroup euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexGroup euiFlexGroup--directionColumn euiFlexGroup--responsive"
          id="wz-dev-tools-editor--left-column"
        >
          <div
            class="euiFlexGroup euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
            id="wz-dev-tools-buttons"
          >
            <svg
              aria-hidden="true"
              aria-labelledby="i1f11de00-87f4-11f0-92ea-ed24b11584c8"
              class="euiIcon euiIcon--medium euiIcon--success euiIcon-isLoading"
              focusable="false"
              height="16"
              id="wz-dev-tools-buttons--send-request"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="i1f11de00-87f4-11f0-92ea-ed24b11584c8"
              >
                Send request
              </title>
              <path
                d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
              />
            </svg>
            <a
              href=""
              id="wz-dev-tools-buttons--go-to-api-reference"
              target="__blank"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                />
              </svg>
            </a>
          </div>
          <textarea
            id="wz-dev-tools-editor--input"
            style="display: none;"
          />
        </div>
        <div
          aria-orientation="vertical"
          id="wz-dev-tools-editor--separator"
          role="separator"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon-isLoaded"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2.5c0-.276.232-.5.5-.5.276 0 .5.229.5.5v11c0 .276-.232.5-.5.5a.503.503 0 0 1-.5-.5v-11Zm3 0c0-.276.232-.5.5-.5.276 0 .5.229.5.5v11c0 .276-.232.5-.5.5a.503.503 0 0 1-.5-.5v-11Z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="euiFlexGroup euiFlexGroup--directionColumn euiFlexGroup--responsive"
          id="wz-dev-tools-editor--right-column"
          style="flex-grow: 1;"
        >
          <textarea
            id="wz-dev-tools-editor--output"
            style="display: none;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
